src/Act/Bounds.hs

```diff
@@ -37,10 +38,15 @@ addBoundsConstructor ctor@(Constructor _ (Interface _ decls) _ pre post invs sta
       pre' = pre
              <> mkCallDataBounds decls
              <> mkEthEnvBounds (ethEnvFromConstructor ctor)
-             <> mkStorageBoundsLoc (nub $ concatMap locsFromExp pre <> concatMap locsFromUpdateRHS stateUpdates)
+             <> mkLocationBounds nonlocalLocs
```

Comment:
Nice workaround! It's worth adding a comment that soundness of this relies on the fact that constructors in Act cannot change non-local fields directly.